src/wasm-builder.h

```diff
@@ -658,6 +670,49 @@ class Builder {
     ret->finalize(type);
     return ret;
   }
+  Try* makeTry(Name name,
+               Expression* body,
+               const std::vector<Name>& catchEvents,
+               const std::vector<Expression*>& catchBodies,
+               Type type) {
+    auto* ret = wasm.allocator.alloc<Try>();
+    ret->name = name;
+    ret->body = body;
+    ret->catchEvents.set(catchEvents);
+    ret->catchBodies.set(catchBodies);
+    ret->finalize(type);
+    return ret;
+  }
+  Try* makeTry(Expression* body, Name delegateTarget) {
+    auto* ret = wasm.allocator.alloc<Try>();
+    ret->body = body;
+    ret->delegateTarget = delegateTarget;
+    ret->finalize();
+    return ret;
+  }
+  Try* makeTry(Name name, Expression* body, Name delegateTarget) {
+    auto* ret = wasm.allocator.alloc<Try>();
+    ret->name = name;
+    ret->body = body;
+    ret->delegateTarget = delegateTarget;
+    ret->finalize();
+    return ret;
+  }
+  Try* makeTry(Expression* body, Name delegateTarget, Type type) {
+    auto* ret = wasm.allocator.alloc<Try>();
+    ret->body = body;
+    ret->delegateTarget = delegateTarget;
+    ret->finalize(type);
+    return ret;
+  }
+  Try* makeTry(Name name, Expression* body, Name delegateTarget, Type type) {
+    auto* ret = wasm.allocator.alloc<Try>();
+    ret->name = name;
+    ret->body = body;
+    ret->delegateTarget = delegateTarget;
+    ret->finalize(type);
+    return ret;
+  }
```

Comment:
This ends up with duplication in all the variants of `makeTry`. How about one main method that is called by the others? Then each variant can be a single line.
```cpp
private:

Try* makeTry(Name name,
             Expression* body,
             const std::vector<Name>& catchEvents,
             const std::vector<Expression*>& catchBodies,
             Name delegateTarget,
             Type type,
             bool hasType) { // differentiate whether a type was passed in
  auto* ret = wasm.allocator.alloc<Try>();
  ret->name = name;
  ret->body = body;
  ret->catchEvents.set(catchEvents);
  ret->catchBodies.set(catchBodies);
  if (hasType) {
    ret->finalize(type);
  } else {
    ret->finalize();
  }
  return ret;
}

public:

Try* makeTry(Expression* body,
             const std::vector<Name>& catchEvents,
             const std::vector<Expression*>& catchBodies) {
  return makeTry(Name(), body, catchEvents, catchBodies, Name(), Type::none, false);
}

Try* makeTry(Expression* body,
             const std::vector<Name>& catchEvents,
             const std::vector<Expression*>& catchBodies,
             Type type) {
  return makeTry(Name(), body, catchEvents, catchBodies, Name(), type, true);
}

[..]

Try* makeTry(Name name, Expression* body, Name delegateTarget, Type type) {
  return makeTry(name, body, {}, {}, delegateTarget, type, true);
}
```

Reply:
Thanks! Done.